File: HILOS_METODOS/ScheduledExecutor.py
import schedule
import time
import threading

class ScheduledExecutor:
    def tarea(self):
            print(f"Tarea ejecutada en: {time.time()}")

    def shutdown(self):
            print("Finalizar el scheduler...")
    global ejecutando
    ejecutando = False

schedule.every(2).seconds.do(tarea)

timer = threading.Timer(10, shutdown)
timer.start()

ejecutando = True
while ejecutando:
    schedule.run_pending()
    time.sleep(1)

File: Hilos Java/HILOS_METODOS/ScheduledExecutor.java
import java.util.concurrent.Executors;
import java.util.concurrent.ScheduledExecutorService;
import java.util.concurrent.TimeUnit;

public class ScheduledExecutor {
    
public static void main(String[] args) {
    
    ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);

    Runnable tarea = () -> System.out.println( "Tarea ejecutada en: " + System.currentTimeMillis());

    scheduler.scheduleAtFixedRate(tarea, 0 , 2, TimeUnit.SECONDS);

    scheduler.schedule(()->{
        System.out.println("Finalizar el scheduler...");

        scheduler.shutdown();
    }, 10, TimeUnit.SECONDS);
}
}
